src/lib.rs

```diff
@@ -110,17 +107,10 @@ impl Infer {
 
     /// Returns the file type of the buffer.
     ///
-    /// # Examples
-    ///
-    /// ```rust
-    /// let info = infer::Infer::new();
-    /// let v = vec![0xFF, 0xD8, 0xFF, 0xAA];
-    /// assert_eq!("image/jpeg", info.get(&v).unwrap().mime);
-    /// assert_eq!("jpg", info.get(&v).unwrap().ext);
-    /// ```
+    /// See [`get`](./fn.get.html).
     pub fn get(&self, buf: &[u8]) -> Option<Type> {
         for (_, mime, ext, matcher) in self.iter_matchers() {
-            if matcher(buf) {
+            if matcher.0(buf) {
```

Comment:
I think having a function on `WrapMatcher` called like `matches(buf: &[u8]) -> bool` would make this code a bit clearer. What do you think?

Reply:
ops didn't want to send a github suggestion

Reply:
Indeed I am not a fan of the WrapMatcher solution but this is all I have for having a `conf fn new()`. Maybe we could address this in the work you are planing to use a `struct` instead of a `tuple` here ?

Reply:
Sure, I can do that there. Since I already wrote it I'll have to rebase it anyway to make it use `WrapMatcher` https://github.com/paolobarbolini/infer/tree/refactor-1 so no problem.